include/types.h

```diff
@@ -183,9 +185,117 @@ struct unified_site {
+    bool operator<(const unified_site& rhs) const noexcept{ return pos < rhs.pos; }
+    bool operator<=(const unified_site& rhs) const noexcept{ return pos <= rhs.pos; }
+
     unified_site(const range& pos_) noexcept : pos(pos_) {}
 };
 
+class loss_stats {
+
+public:
+
+    // Constructor takes a sample name; loss_stats should be
+    // uniquely associated with each sample
+    loss_stats(const std::string sample_name_) noexcept : sample_name(sample_name_) {}
+
+    // Prepares the map with the given unified_site, requires that this method
+    // is not called again when an entry for the site exists
+    Status initializeSite(const unified_site site) noexcept {
+        if (orig_calls_for_sites.find(site) != orig_calls_for_sites.end())
+            return Status::Failure("loss_stats: initializeSite called for existing site");
+
+        orig_calls_for_sites.insert(std::make_pair(site, std::map<range, int>() ));
+        return Status::OK();
+    }
+
+    // Called for each bcf record that is associated with the unified_site
+    // examined to update "denominator" of total calls/bp covered in orig
+    // dataset
+    Status addCallForSite(const unified_site site, const range call, int n_calls) noexcept {
+        if (orig_calls_for_sites.find(site) == orig_calls_for_sites.end())
+            return Status::Failure("loss_stats: addCallForSite called before initializeSite or after finalizeLossForSite.");
+
+        auto& orig_calls_for_site = orig_calls_for_sites[site];
+
+        auto call_within_site_p = call.intersect(site.pos);
+
+        if (call_within_site_p) {
+            range call_within_site = *call_within_site_p;
+            orig_calls_for_site[call_within_site] += n_calls;
+            n_calls_total += n_calls;
+            n_bp_total += n_calls * (call_within_site.size());
+        }
+        return Status::OK();
+    }
+
+    // Called after joint genotyping for a unified_site to update the loss
+    // associated with that site.
+    Status finalizeLossForSite(const unified_site site, int n_no_calls) noexcept {
+        if (orig_calls_for_sites.find(site) == orig_calls_for_sites.end())
+            return Status::Failure("loss_stats: finalizeLossForSite called before initializeSite or after finalizeLossForSite.");
+
+        if (n_no_calls) {
+            n_no_calls_total += n_no_calls;
+
+            auto& calls_for_site = orig_calls_for_sites[site];
+            for (auto& kv : calls_for_site) {
+                // calls_lost will be 0 if n_orig_calls and n_no_calls are both 1 (which is interpreted as no loss of calls)
+                int n_calls_lost_for_site = (kv.second * n_no_calls) / 2;
+                n_calls_lost += n_calls_lost_for_site;
+
+                // assumes that range in calls_for_site has already been
+                // restricted to intersection with site
+                n_bp_lost += kv.first.size();
```

Comment:
I confess not really following the calculations here- above we're multiplying by `n_no_calls` and then dividing by 2, here we are doing neither. A little more explanation would be helpful.